core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -215,6 +218,15 @@ namespace kagome::consensus {
     }
   }
 
+  void BabeImpl::doOnSynchronized(std::function<void()> handler) {
+    on_synchronized_ = std::move(handler);
+    if (current_state_ == State::SYNCHRONIZED) {
+      if (auto on_synchronized = std::move(on_synchronized_)) {
+        on_synchronized();
+      }
+    }
```

Comment:
```c++
if (current_state_ == State::SYNCHRONIZED) {
  if (handler)
    handler();
} else
  on_synchronized_ = std::move(handler);

Reply:
Just another way.

Reply:
It makes the same with a less ```move``` calls(optimizer will fix it, but it doesn't mean that we can make extra code because we it alowes it to us :) ) More over, when I saw this construction, the first idea was that you made this to replace the existed callback in ```on_synchronized_```. And I had to check it additionaly and saw that ```on_synchronized_``` can not be not null in ```SYNCHRONIZED``` state.

Reply:
I think I am more inclined towards @iceseer 's suggestion, it's somewhat easier to read.